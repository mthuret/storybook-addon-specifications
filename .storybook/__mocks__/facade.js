const { mocks } = require('../../src');
module.exports = {
  ...mocks,
  storiesOf: mocks.snapshotStoriesOf,
};
