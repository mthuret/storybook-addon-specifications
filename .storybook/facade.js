if (!global['storiesOf']) {
  var {
    storiesOf: storiesOfReal,
    action: actionReal,
    linkTo: linkToReal,
  } = require("@kadira/storybook");
}

if (!global['specs']) {
  var {
    specs: specsReal,
  } = require('../dist');
}

export const storiesOf = global['storiesOf'] || storiesOfReal;
export const action = global['action'] || actionReal;
export const linkTo = global['linkTo'] || linkToReal;
export const specs = global['specs'] || specsReal;
