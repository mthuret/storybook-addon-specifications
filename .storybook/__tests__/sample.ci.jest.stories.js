import React from "react";
import {storiesOf, action, specs} from "../facade";
import {mount} from "enzyme";
import expect from "expect";


const stories = storiesOf('Button - CI JEST Sample', module);

stories.add('Hello World', function () {
  const helloWorldStory =
    <button onClick={action('Hello World')}>
      Hello World
    </button>;

  specs(() => describe('Hello World', function () {
    let output;
    beforeEach(function() {
      console.log('BEFORE EACH');
      output = mount(helloWorldStory);
    });

    afterEach(function() {
      console.log('AFTER EACH');
    });

    it('Should have the Hello World label', function () {
      expect(output.text()).toContain('Hello World');
    });

    it('Should have the Hello World label', function () {
      expect(output.text()).toContain('Hello World');
    });

  }));
  return helloWorldStory;
});

stories.add('Hello World Streamlined', function () {
  const helloWorldStory =
    <button onClick={action('Hello World')}>
      Hello World
    </button>;

  describe('Hello World Streamlined', function () {
    let output;
    beforeEach(function() {
      console.log('BEFORE EACH');
      output = mount(helloWorldStory);
    });

    afterEach(function() {
      console.log('AFTER EACH');
    });

    it('Should have the Hello World label', function () {
      expect(output.text()).toContain('Hello World');
    });

    it('Should have the Hello World label', function () {
      expect(output.text()).toContain('Hello World');
    });
  });

  return helloWorldStory;
});


stories.add('Hello Earth', function () {
  const helloEarthStory =
    <button onClick={action('Hello Earth')}>
      Hello Earth
    </button>;

  specs(() => describe('Hello Earth', function () {
    it('Should have the Hello Earth label', function () {
      let output = mount(helloEarthStory);
      expect(output.text()).toContain('Hello Earth');
    });
  }));

  return helloEarthStory;
});
