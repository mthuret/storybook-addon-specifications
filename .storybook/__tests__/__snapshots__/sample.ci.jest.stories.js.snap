exports[`test Hello Earth 1`] = `
<button
  onClick={[Function NO_FUNC]}>
  Hello Earth
</button>
`;

exports[`test Hello World 1`] = `
<button
  onClick={[Function NO_FUNC]}>
  Hello World
</button>
`;

exports[`test Hello World Streamlined 1`] = `
<button
  onClick={[Function NO_FUNC]}>
  Hello World
</button>
`;
